src/lib/operators/aggregate_sort.cpp

```diff
@@ -554,12 +559,14 @@ std::shared_ptr<const Table> AggregateSort::_on_execute() {
    * Write outputted values into the result table
    */
   const auto write_groupby_column = [&](const ColumnID input_column_id, const ColumnID output_column_id) {
+    const auto column_is_nullable = _output_column_definitions.at(output_column_id).nullable;
     auto group_boundary_iter = group_boundaries.cbegin();
     auto data_type = input_table->column_data_type(input_column_id);
     resolve_data_type(data_type, [&](auto type) {
       using ColumnDataType = typename decltype(type)::type;
       auto values = pmr_vector<ColumnDataType>(group_boundaries.size() + 1);
-      auto null_values = pmr_vector<bool>(group_boundaries.size() + 1);
+      pmr_vector<bool> null_values;
+      if (column_is_nullable) null_values = pmr_vector<bool>(group_boundaries.size() + 1);
```

Comment:
Personally, I think that a ternary operator is slightly easier to read, but that might be just my opinion. No need to change.

Reply:
what would be the else part of the ternary operator in this case? am i missing something?

Reply:
`auto null_values = pmr_vector<bool>(column_is_nullable ? group_boundaries.size() + 1 : 0);`